From 0ed101e0152476aa8891b10f47574628d929f3ce Mon Sep 17 00:00:00 2001
From: Manos Koukoutos <manoskouk@chromium.org>
Date: Thu, 7 Jul 2022 11:52:08 +0000
Subject: [PATCH] Remove some unused includes (4)

Mostly src/codegen, src/compiler, src/interpreter, src/libplatform.

Drive-by: Remove some unreachable code.

Bug: v8:13006
Change-Id: I1a9467f7e42531c545f660d35416c388e8ef9d3c
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3749193
Commit-Queue: Manos Koukoutos <manoskouk@chromium.org>
Reviewed-by: Toon Verwaest <verwaest@chromium.org>
Cr-Commit-Position: refs/heads/main@{#81613}
---
 src/codegen/arm/assembler-arm-inl.h                   |  2 +-
 src/codegen/arm64/assembler-arm64-inl.h               |  1 +
 src/codegen/arm64/cpu-arm64.cc                        |  1 +
 src/codegen/ia32/assembler-ia32-inl.h                 |  4 ++--
 src/codegen/loong64/assembler-loong64.cc              |  1 +
 src/codegen/mips/assembler-mips.cc                    |  1 +
 src/codegen/mips64/assembler-mips64.cc                |  1 +
 src/codegen/ppc/assembler-ppc-inl.h                   |  4 ++--
 src/codegen/reloc-info.h                              |  1 -
 src/codegen/riscv64/assembler-riscv64.cc              |  1 +
 src/codegen/s390/assembler-s390-inl.h                 |  4 ++--
 src/codegen/x64/assembler-x64-inl.h                   |  4 ++--
 src/compiler/js-operator.h                            |  1 +
 src/compiler/machine-graph-verifier.h                 |  2 --
 src/compiler/machine-graph.cc                         |  1 -
 src/compiler/machine-operator-reducer.cc              |  1 -
 src/compiler/map-inference.h                          |  2 --
 src/compiler/memory-lowering.cc                       |  1 -
 src/compiler/memory-optimizer.cc                      |  2 --
 src/compiler/node-origin-table.h                      |  2 --
 src/compiler/node-properties.cc                       |  5 +----
 src/compiler/node-properties.h                        |  2 --
 src/compiler/operation-typer.cc                       |  5 ++---
 src/compiler/operation-typer.h                        |  1 -
 src/compiler/operator-properties.h                    |  1 -
 src/compiler/operator.h                               |  1 -
 src/compiler/osr.cc                                   |  4 +---
 src/compiler/osr.h                                    |  2 +-
 src/compiler/per-isolate-compiler-cache.h             |  1 -
 src/compiler/pipeline-statistics.cc                   |  1 -
 src/compiler/pipeline.cc                              |  5 +----
 src/compiler/pipeline.h                               |  1 -
 src/compiler/processed-feedback.h                     |  1 -
 src/compiler/property-access-builder.cc               |  5 ++---
 src/compiler/property-access-builder.h                |  1 -
 src/compiler/raw-machine-assembler.cc                 |  1 -
 src/compiler/raw-machine-assembler.h                  |  1 -
 src/compiler/refs-map.h                               |  1 -
 src/compiler/representation-change.cc                 |  2 --
 src/compiler/scheduler.h                              |  3 ---
 src/compiler/select-lowering.cc                       |  2 --
 src/compiler/simplified-lowering.cc                   |  4 +---
 src/compiler/simplified-operator-reducer.cc           |  1 -
 src/compiler/simplified-operator-reducer.h            |  1 -
 src/compiler/simplified-operator.cc                   |  3 +--
 src/compiler/simplified-operator.h                    |  1 -
 src/compiler/state-values-utils.cc                    |  1 -
 src/compiler/state-values-utils.h                     |  1 -
 src/compiler/store-store-elimination.cc               |  1 -
 src/compiler/type-cache.h                             |  1 -
 src/compiler/typed-optimization.h                     |  1 -
 src/compiler/typer.cc                                 |  1 -
 src/compiler/typer.h                                  |  1 -
 src/compiler/types.cc                                 |  3 +--
 src/compiler/value-numbering-reducer.cc               |  1 -
 src/compiler/verifier.cc                              |  2 --
 src/compiler/verifier.h                               |  1 -
 src/compiler/wasm-compiler.cc                         | 11 +----------
 src/compiler/wasm-compiler.h                          |  1 -
 src/compiler/wasm-escape-analysis.cc                  |  2 +-
 src/compiler/wasm-gc-lowering.cc                      |  2 --
 src/compiler/wasm-inlining.h                          |  2 +-
 src/compiler/wasm-loop-peeling.cc                     |  3 ---
 src/compiler/wasm-loop-peeling.h                      |  3 ++-
 src/compiler/wasm-typer.cc                            |  1 -
 src/compiler/zone-stats.h                             |  1 -
 src/init/setup-isolate-full.cc                        |  4 +---
 src/interpreter/bytecode-array-builder.h              |  2 --
 src/interpreter/bytecode-array-iterator.h             |  1 -
 src/interpreter/bytecode-array-writer.cc              |  1 -
 src/interpreter/bytecode-array-writer.h               |  1 -
 src/interpreter/bytecode-decoder.h                    |  2 --
 src/interpreter/bytecode-flags.cc                     |  1 -
 src/interpreter/bytecode-generator.cc                 |  4 +---
 src/interpreter/bytecode-generator.h                  |  1 -
 src/interpreter/bytecode-node.cc                      |  1 -
 src/interpreter/bytecode-node.h                       |  1 -
 src/interpreter/bytecode-register-allocator.h         |  2 --
 src/interpreter/bytecode-register-optimizer.h         |  2 ++
 src/interpreter/bytecode-source-info.cc               |  1 -
 src/interpreter/bytecodes.cc                          |  1 -
 src/interpreter/constant-array-builder.cc             |  2 +-
 src/interpreter/constant-array-builder.h              |  3 +--
 src/interpreter/handler-table-builder.h               |  1 -
 src/interpreter/interpreter-assembler.cc              |  2 --
 src/interpreter/interpreter-assembler.h               |  2 --
 src/interpreter/interpreter-generator.cc              |  3 ---
 src/interpreter/interpreter-intrinsics-generator.cc   |  5 -----
 src/interpreter/interpreter.cc                        |  4 ----
 src/interpreter/interpreter.h                         |  2 +-
 src/json/json-parser.h                                |  1 -
 src/json/json-stringifier.cc                          |  1 -
 src/libplatform/default-foreground-task-runner.cc     |  2 +-
 src/libplatform/default-platform.cc                   |  1 -
 src/libplatform/default-platform.h                    |  2 --
 src/libplatform/delayed-task-queue.h                  |  1 -
 src/libplatform/task-queue.h                          |  1 -
 src/libplatform/worker-thread.h                       |  1 -
 src/wasm/jump-table-assembler.h                       |  1 +
 test/cctest/compiler/code-assembler-tester.h          |  3 ++-
 test/cctest/compiler/test-codegen.h                   |  6 ++----
 .../compiler/test-run-calls-to-external-references.cc |  2 +-
 test/common/assembler-tester.h                        |  4 ++--
 103 files changed, 48 insertions(+), 160 deletions(-)

diff --git a/src/codegen/arm/assembler-arm-inl.h b/src/codegen/arm/assembler-arm-inl.h
index 9080b3e0b3..dbaff9e6fc 100644
--- a/src/codegen/arm/assembler-arm-inl.h
+++ b/src/codegen/arm/assembler-arm-inl.h
@@ -38,8 +38,8 @@
 #define V8_CODEGEN_ARM_ASSEMBLER_ARM_INL_H_
 
 #include "src/codegen/arm/assembler-arm.h"
-
 #include "src/codegen/assembler.h"
+#include "src/codegen/flush-instruction-cache.h"
 #include "src/debug/debug.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/smi.h"
diff --git a/src/codegen/arm64/assembler-arm64-inl.h b/src/codegen/arm64/assembler-arm64-inl.h
index 0f42f0efa9..da4201b283 100644
--- a/src/codegen/arm64/assembler-arm64-inl.h
+++ b/src/codegen/arm64/assembler-arm64-inl.h
@@ -10,6 +10,7 @@
 #include "src/base/memory.h"
 #include "src/codegen/arm64/assembler-arm64.h"
 #include "src/codegen/assembler.h"
+#include "src/codegen/flush-instruction-cache.h"
 #include "src/debug/debug.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/smi.h"
diff --git a/src/codegen/arm64/cpu-arm64.cc b/src/codegen/arm64/cpu-arm64.cc
index 96bb73f0d4..997a8ea044 100644
--- a/src/codegen/arm64/cpu-arm64.cc
+++ b/src/codegen/arm64/cpu-arm64.cc
@@ -8,6 +8,7 @@
 
 #include "src/codegen/arm64/utils-arm64.h"
 #include "src/codegen/cpu-features.h"
+#include "src/codegen/flush-instruction-cache.h"
 
 #if V8_OS_DARWIN
 #include <libkern/OSCacheControl.h>
diff --git a/src/codegen/ia32/assembler-ia32-inl.h b/src/codegen/ia32/assembler-ia32-inl.h
index 607c0aca6e..41dfde3393 100644
--- a/src/codegen/ia32/assembler-ia32-inl.h
+++ b/src/codegen/ia32/assembler-ia32-inl.h
@@ -37,10 +37,10 @@
 #ifndef V8_CODEGEN_IA32_ASSEMBLER_IA32_INL_H_
 #define V8_CODEGEN_IA32_ASSEMBLER_IA32_INL_H_
 
-#include "src/codegen/ia32/assembler-ia32.h"
-
 #include "src/base/memory.h"
 #include "src/codegen/assembler.h"
+#include "src/codegen/flush-instruction-cache.h"
+#include "src/codegen/ia32/assembler-ia32.h"
 #include "src/debug/debug.h"
 #include "src/objects/objects-inl.h"
 
diff --git a/src/codegen/loong64/assembler-loong64.cc b/src/codegen/loong64/assembler-loong64.cc
index 3e390e544a..e10b5bd194 100644
--- a/src/codegen/loong64/assembler-loong64.cc
+++ b/src/codegen/loong64/assembler-loong64.cc
@@ -7,6 +7,7 @@
 #if V8_TARGET_ARCH_LOONG64
 
 #include "src/base/cpu.h"
+#include "src/codegen/flush-instruction-cache.h"
 #include "src/codegen/loong64/assembler-loong64-inl.h"
 #include "src/codegen/machine-type.h"
 #include "src/codegen/safepoint-table.h"
diff --git a/src/codegen/mips/assembler-mips.cc b/src/codegen/mips/assembler-mips.cc
index 788651e6fc..63f0373b68 100644
--- a/src/codegen/mips/assembler-mips.cc
+++ b/src/codegen/mips/assembler-mips.cc
@@ -38,6 +38,7 @@
 
 #include "src/base/bits.h"
 #include "src/base/cpu.h"
+#include "src/codegen/flush-instruction-cache.h"
 #include "src/codegen/mips/assembler-mips-inl.h"
 #include "src/codegen/safepoint-table.h"
 #include "src/codegen/string-constants.h"
diff --git a/src/codegen/mips64/assembler-mips64.cc b/src/codegen/mips64/assembler-mips64.cc
index 14f3734812..f32d1479cb 100644
--- a/src/codegen/mips64/assembler-mips64.cc
+++ b/src/codegen/mips64/assembler-mips64.cc
@@ -37,6 +37,7 @@
 #if V8_TARGET_ARCH_MIPS64
 
 #include "src/base/cpu.h"
+#include "src/codegen/flush-instruction-cache.h"
 #include "src/codegen/machine-type.h"
 #include "src/codegen/mips64/assembler-mips64-inl.h"
 #include "src/codegen/safepoint-table.h"
diff --git a/src/codegen/ppc/assembler-ppc-inl.h b/src/codegen/ppc/assembler-ppc-inl.h
index 364b20e596..6416fce2f6 100644
--- a/src/codegen/ppc/assembler-ppc-inl.h
+++ b/src/codegen/ppc/assembler-ppc-inl.h
@@ -37,9 +37,9 @@
 #ifndef V8_CODEGEN_PPC_ASSEMBLER_PPC_INL_H_
 #define V8_CODEGEN_PPC_ASSEMBLER_PPC_INL_H_
 
-#include "src/codegen/ppc/assembler-ppc.h"
-
 #include "src/codegen/assembler.h"
+#include "src/codegen/flush-instruction-cache.h"
+#include "src/codegen/ppc/assembler-ppc.h"
 #include "src/debug/debug.h"
 #include "src/objects/objects-inl.h"
 
diff --git a/src/codegen/reloc-info.h b/src/codegen/reloc-info.h
index 3b0b05213f..98adfb97c5 100644
--- a/src/codegen/reloc-info.h
+++ b/src/codegen/reloc-info.h
@@ -5,7 +5,6 @@
 #ifndef V8_CODEGEN_RELOC_INFO_H_
 #define V8_CODEGEN_RELOC_INFO_H_
 
-#include "src/codegen/flush-instruction-cache.h"
 #include "src/common/globals.h"
 #include "src/objects/code.h"
 
diff --git a/src/codegen/riscv64/assembler-riscv64.cc b/src/codegen/riscv64/assembler-riscv64.cc
index f01a726ebb..a661be0ba0 100644
--- a/src/codegen/riscv64/assembler-riscv64.cc
+++ b/src/codegen/riscv64/assembler-riscv64.cc
@@ -37,6 +37,7 @@
 #include "src/codegen/riscv64/assembler-riscv64.h"
 
 #include "src/base/cpu.h"
+#include "src/codegen/flush-instruction-cache.h"
 #include "src/codegen/riscv64/assembler-riscv64-inl.h"
 #include "src/codegen/safepoint-table.h"
 #include "src/codegen/string-constants.h"
diff --git a/src/codegen/s390/assembler-s390-inl.h b/src/codegen/s390/assembler-s390-inl.h
index 8170c02204..a29bf79e25 100644
--- a/src/codegen/s390/assembler-s390-inl.h
+++ b/src/codegen/s390/assembler-s390-inl.h
@@ -37,9 +37,9 @@
 #ifndef V8_CODEGEN_S390_ASSEMBLER_S390_INL_H_
 #define V8_CODEGEN_S390_ASSEMBLER_S390_INL_H_
 
-#include "src/codegen/s390/assembler-s390.h"
-
 #include "src/codegen/assembler.h"
+#include "src/codegen/flush-instruction-cache.h"
+#include "src/codegen/s390/assembler-s390.h"
 #include "src/debug/debug.h"
 #include "src/objects/objects-inl.h"
 
diff --git a/src/codegen/x64/assembler-x64-inl.h b/src/codegen/x64/assembler-x64-inl.h
index dd74e93f10..398ff6d19b 100644
--- a/src/codegen/x64/assembler-x64-inl.h
+++ b/src/codegen/x64/assembler-x64-inl.h
@@ -5,10 +5,10 @@
 #ifndef V8_CODEGEN_X64_ASSEMBLER_X64_INL_H_
 #define V8_CODEGEN_X64_ASSEMBLER_X64_INL_H_
 
-#include "src/codegen/x64/assembler-x64.h"
-
 #include "src/base/cpu.h"
 #include "src/base/memory.h"
+#include "src/codegen/flush-instruction-cache.h"
+#include "src/codegen/x64/assembler-x64.h"
 #include "src/debug/debug.h"
 #include "src/objects/objects-inl.h"
 
diff --git a/src/compiler/js-operator.h b/src/compiler/js-operator.h
index 2d7a90dc9e..12408aea6e 100644
--- a/src/compiler/js-operator.h
+++ b/src/compiler/js-operator.h
@@ -13,6 +13,7 @@
 #include "src/compiler/node-properties.h"
 #include "src/compiler/node.h"
 #include "src/compiler/opcodes.h"
+#include "src/objects/feedback-cell.h"
 #include "src/runtime/runtime.h"
 
 namespace v8 {
diff --git a/src/compiler/machine-graph-verifier.h b/src/compiler/machine-graph-verifier.h
index 26e5d772c2..656f7d575b 100644
--- a/src/compiler/machine-graph-verifier.h
+++ b/src/compiler/machine-graph-verifier.h
@@ -5,8 +5,6 @@
 #ifndef V8_COMPILER_MACHINE_GRAPH_VERIFIER_H_
 #define V8_COMPILER_MACHINE_GRAPH_VERIFIER_H_
 
-#include "src/base/macros.h"
-
 namespace v8 {
 namespace internal {
 class Zone;
diff --git a/src/compiler/machine-graph.cc b/src/compiler/machine-graph.cc
index 34464cfb05..0822b27847 100644
--- a/src/compiler/machine-graph.cc
+++ b/src/compiler/machine-graph.cc
@@ -5,7 +5,6 @@
 #include "src/compiler/machine-graph.h"
 
 #include "src/codegen/external-reference.h"
-#include "src/compiler/node-properties.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/machine-operator-reducer.cc b/src/compiler/machine-operator-reducer.cc
index fed4feda9a..a676063379 100644
--- a/src/compiler/machine-operator-reducer.cc
+++ b/src/compiler/machine-operator-reducer.cc
@@ -12,7 +12,6 @@
 #include "src/base/ieee754.h"
 #include "src/base/logging.h"
 #include "src/base/overflowing-math.h"
-#include "src/codegen/tnode.h"
 #include "src/compiler/diamond.h"
 #include "src/compiler/graph.h"
 #include "src/compiler/js-operator.h"
diff --git a/src/compiler/map-inference.h b/src/compiler/map-inference.h
index e1392b6805..6d84035d4b 100644
--- a/src/compiler/map-inference.h
+++ b/src/compiler/map-inference.h
@@ -5,7 +5,6 @@
 #ifndef V8_COMPILER_MAP_INFERENCE_H_
 #define V8_COMPILER_MAP_INFERENCE_H_
 
-#include "include/v8config.h"
 #include "src/compiler/graph-reducer.h"
 #include "src/objects/instance-type.h"
 #include "src/objects/map.h"
@@ -13,7 +12,6 @@
 namespace v8 {
 namespace internal {
 
-
 namespace compiler {
 
 class CompilationDependencies;
diff --git a/src/compiler/memory-lowering.cc b/src/compiler/memory-lowering.cc
index 5d6b49abfd..07b2a5b259 100644
--- a/src/compiler/memory-lowering.cc
+++ b/src/compiler/memory-lowering.cc
@@ -5,7 +5,6 @@
 #include "src/compiler/memory-lowering.h"
 
 #include "src/codegen/interface-descriptors-inl.h"
-#include "src/compiler/access-builder.h"
 #include "src/compiler/js-graph.h"
 #include "src/compiler/linkage.h"
 #include "src/compiler/node-matchers.h"
diff --git a/src/compiler/memory-optimizer.cc b/src/compiler/memory-optimizer.cc
index e74e80c4e4..dc0d80ff00 100644
--- a/src/compiler/memory-optimizer.cc
+++ b/src/compiler/memory-optimizer.cc
@@ -5,11 +5,9 @@
 #include "src/compiler/memory-optimizer.h"
 
 #include "src/base/logging.h"
-#include "src/codegen/interface-descriptors.h"
 #include "src/codegen/tick-counter.h"
 #include "src/compiler/js-graph.h"
 #include "src/compiler/linkage.h"
-#include "src/compiler/node-matchers.h"
 #include "src/compiler/node-properties.h"
 #include "src/compiler/node.h"
 #include "src/roots/roots-inl.h"
diff --git a/src/compiler/node-origin-table.h b/src/compiler/node-origin-table.h
index 899c62d411..fa63079b6c 100644
--- a/src/compiler/node-origin-table.h
+++ b/src/compiler/node-origin-table.h
@@ -8,8 +8,6 @@
 #include <limits>
 
 #include "src/base/compiler-specific.h"
-#include "src/codegen/source-position.h"
-#include "src/common/globals.h"
 #include "src/compiler/node-aux-data.h"
 
 namespace v8 {
diff --git a/src/compiler/node-properties.cc b/src/compiler/node-properties.cc
index 98891a5924..051eeeb5ef 100644
--- a/src/compiler/node-properties.cc
+++ b/src/compiler/node-properties.cc
@@ -3,18 +3,15 @@
 // found in the LICENSE file.
 
 #include "src/compiler/node-properties.h"
+
 #include "src/compiler/common-operator.h"
 #include "src/compiler/graph.h"
 #include "src/compiler/js-heap-broker.h"
-#include "src/compiler/js-operator.h"
-#include "src/compiler/linkage.h"
 #include "src/compiler/map-inference.h"
 #include "src/compiler/node-matchers.h"
 #include "src/compiler/operator-properties.h"
 #include "src/compiler/simplified-operator.h"
 #include "src/compiler/verifier.h"
-#include "src/handles/handles-inl.h"
-#include "src/objects/objects-inl.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/node-properties.h b/src/compiler/node-properties.h
index 766f16bfac..f56aa6ebfb 100644
--- a/src/compiler/node-properties.h
+++ b/src/compiler/node-properties.h
@@ -10,8 +10,6 @@
 #include "src/compiler/node.h"
 #include "src/compiler/operator-properties.h"
 #include "src/compiler/types.h"
-#include "src/objects/map.h"
-#include "src/zone/zone-handle-set.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/operation-typer.cc b/src/compiler/operation-typer.cc
index ed51db6a11..267b5677e5 100644
--- a/src/compiler/operation-typer.cc
+++ b/src/compiler/operation-typer.cc
@@ -9,9 +9,8 @@
 #include "src/compiler/type-cache.h"
 #include "src/compiler/types.h"
 #include "src/execution/isolate.h"
-#include "src/heap/factory.h"
-
-#include "src/objects/objects-inl.h"
+#include "src/heap/factory-inl.h"
+#include "src/objects/oddball.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/operation-typer.h b/src/compiler/operation-typer.h
index 728e297a1b..24dd0413c1 100644
--- a/src/compiler/operation-typer.h
+++ b/src/compiler/operation-typer.h
@@ -8,7 +8,6 @@
 #include "src/base/flags.h"
 #include "src/compiler/opcodes.h"
 #include "src/compiler/types.h"
-#include "src/objects/objects.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/operator-properties.h b/src/compiler/operator-properties.h
index e566a58a02..708cd7db1f 100644
--- a/src/compiler/operator-properties.h
+++ b/src/compiler/operator-properties.h
@@ -6,7 +6,6 @@
 #define V8_COMPILER_OPERATOR_PROPERTIES_H_
 
 #include "src/base/macros.h"
-#include "src/common/globals.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/operator.h b/src/compiler/operator.h
index f641394eb1..005ac4e040 100644
--- a/src/compiler/operator.h
+++ b/src/compiler/operator.h
@@ -12,7 +12,6 @@
 #include "src/base/functional.h"
 #include "src/common/globals.h"
 #include "src/handles/handles.h"
-#include "src/objects/feedback-cell.h"
 #include "src/zone/zone.h"
 
 namespace v8 {
diff --git a/src/compiler/osr.cc b/src/compiler/osr.cc
index d8ebe23abd..2cfe8a27b5 100644
--- a/src/compiler/osr.cc
+++ b/src/compiler/osr.cc
@@ -6,9 +6,7 @@
 
 #include "src/codegen/optimized-compilation-info.h"
 #include "src/compiler/frame.h"
-#include "src/objects/objects-inl.h"
-#include "src/objects/objects.h"
-#include "src/objects/shared-function-info.h"
+#include "src/objects/js-array-inl.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/osr.h b/src/compiler/osr.h
index 673b29e7ec..91a98f40b1 100644
--- a/src/compiler/osr.h
+++ b/src/compiler/osr.h
@@ -5,7 +5,7 @@
 #ifndef V8_COMPILER_OSR_H_
 #define V8_COMPILER_OSR_H_
 
-#include "src/zone/zone.h"
+#include <stddef.h>
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/per-isolate-compiler-cache.h b/src/compiler/per-isolate-compiler-cache.h
index 716e686921..0a79f6da7c 100644
--- a/src/compiler/per-isolate-compiler-cache.h
+++ b/src/compiler/per-isolate-compiler-cache.h
@@ -7,7 +7,6 @@
 
 #include "src/compiler/refs-map.h"
 #include "src/execution/isolate.h"
-#include "src/zone/zone-containers.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/pipeline-statistics.cc b/src/compiler/pipeline-statistics.cc
index 2ec39e901b..a7ddb8381b 100644
--- a/src/compiler/pipeline-statistics.cc
+++ b/src/compiler/pipeline-statistics.cc
@@ -9,7 +9,6 @@
 #include "src/codegen/optimized-compilation-info.h"
 #include "src/compiler/zone-stats.h"
 #include "src/objects/shared-function-info.h"
-#include "src/objects/string.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/pipeline.cc b/src/compiler/pipeline.cc
index 13719ad763..bde090d679 100644
--- a/src/compiler/pipeline.cc
+++ b/src/compiler/pipeline.cc
@@ -10,13 +10,13 @@
 #include <sstream>
 
 #include "src/base/optional.h"
-#include "src/base/platform/elapsed-timer.h"
 #include "src/builtins/profile-data-reader.h"
 #include "src/codegen/assembler-inl.h"
 #include "src/codegen/bailout-reason.h"
 #include "src/codegen/compiler.h"
 #include "src/codegen/optimized-compilation-info.h"
 #include "src/codegen/register-configuration.h"
+#include "src/codegen/reloc-info.h"
 #include "src/common/high-allocation-throughput-scope.h"
 #include "src/compiler/add-type-assertions-reducer.h"
 #include "src/compiler/backend/code-generator.h"
@@ -93,14 +93,11 @@
 #include "src/diagnostics/disassembler.h"
 #include "src/execution/isolate-inl.h"
 #include "src/heap/local-heap.h"
-#include "src/init/bootstrapper.h"
 #include "src/logging/code-events.h"
 #include "src/logging/counters.h"
 #include "src/logging/runtime-call-stats-scope.h"
 #include "src/objects/shared-function-info.h"
-#include "src/parsing/parse-info.h"
 #include "src/tracing/trace-event.h"
-#include "src/tracing/traced-value.h"
 #include "src/utils/ostreams.h"
 #include "src/utils/utils.h"
 
diff --git a/src/compiler/pipeline.h b/src/compiler/pipeline.h
index deca8070e2..b0ebf4df26 100644
--- a/src/compiler/pipeline.h
+++ b/src/compiler/pipeline.h
@@ -11,7 +11,6 @@
 // Do not include anything from src/compiler here!
 #include "src/common/globals.h"
 #include "src/objects/code.h"
-#include "src/objects/objects.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/processed-feedback.h b/src/compiler/processed-feedback.h
index 1df4c2e071..c5555b0c49 100644
--- a/src/compiler/processed-feedback.h
+++ b/src/compiler/processed-feedback.h
@@ -5,7 +5,6 @@
 #ifndef V8_COMPILER_PROCESSED_FEEDBACK_H_
 #define V8_COMPILER_PROCESSED_FEEDBACK_H_
 
-#include "src/compiler/feedback-source.h"
 #include "src/compiler/heap-refs.h"
 
 namespace v8 {
diff --git a/src/compiler/property-access-builder.cc b/src/compiler/property-access-builder.cc
index b77a9423b8..74f14764e4 100644
--- a/src/compiler/property-access-builder.cc
+++ b/src/compiler/property-access-builder.cc
@@ -11,11 +11,10 @@
 #include "src/compiler/js-graph.h"
 #include "src/compiler/node-matchers.h"
 #include "src/compiler/simplified-operator.h"
-#include "src/execution/isolate-inl.h"
-#include "src/objects/field-index-inl.h"
 #include "src/objects/heap-number.h"
 #include "src/objects/internal-index.h"
-#include "src/objects/lookup.h"
+#include "src/objects/js-function.h"
+#include "src/objects/map-inl.h"
 #include "src/objects/property-details.h"
 
 namespace v8 {
diff --git a/src/compiler/property-access-builder.h b/src/compiler/property-access-builder.h
index bf02ad13b5..439514d7b3 100644
--- a/src/compiler/property-access-builder.h
+++ b/src/compiler/property-access-builder.h
@@ -10,7 +10,6 @@
 #include "src/compiler/js-heap-broker.h"
 #include "src/compiler/node.h"
 #include "src/handles/handles.h"
-#include "src/objects/map.h"
 #include "src/zone/zone-containers.h"
 
 namespace v8 {
diff --git a/src/compiler/raw-machine-assembler.cc b/src/compiler/raw-machine-assembler.cc
index 21f8c184f3..cdd6b4a55a 100644
--- a/src/compiler/raw-machine-assembler.cc
+++ b/src/compiler/raw-machine-assembler.cc
@@ -7,7 +7,6 @@
 #include "src/base/small-vector.h"
 #include "src/compiler/compiler-source-position-table.h"
 #include "src/compiler/node-properties.h"
-#include "src/compiler/pipeline.h"
 #include "src/compiler/scheduler.h"
 #include "src/heap/factory-inl.h"
 
diff --git a/src/compiler/raw-machine-assembler.h b/src/compiler/raw-machine-assembler.h
index d8e7105d34..50fb53e3c9 100644
--- a/src/compiler/raw-machine-assembler.h
+++ b/src/compiler/raw-machine-assembler.h
@@ -8,7 +8,6 @@
 #include <initializer_list>
 #include <type_traits>
 
-#include "src/codegen/assembler.h"
 #include "src/common/globals.h"
 #include "src/compiler/access-builder.h"
 #include "src/compiler/common-operator.h"
diff --git a/src/compiler/refs-map.h b/src/compiler/refs-map.h
index 3b26276072..fec5a4116a 100644
--- a/src/compiler/refs-map.h
+++ b/src/compiler/refs-map.h
@@ -6,7 +6,6 @@
 #define V8_COMPILER_REFS_MAP_H_
 
 #include "src/base/hashmap.h"
-#include "src/common/globals.h"
 #include "src/zone/zone.h"
 
 namespace v8 {
diff --git a/src/compiler/representation-change.cc b/src/compiler/representation-change.cc
index fcefde37e0..84560d5460 100644
--- a/src/compiler/representation-change.cc
+++ b/src/compiler/representation-change.cc
@@ -6,9 +6,7 @@
 
 #include <sstream>
 
-#include "src/base/bits.h"
 #include "src/base/safe_conversions.h"
-#include "src/codegen/code-factory.h"
 #include "src/compiler/js-heap-broker.h"
 #include "src/compiler/machine-operator.h"
 #include "src/compiler/node-matchers.h"
diff --git a/src/compiler/scheduler.h b/src/compiler/scheduler.h
index 15be72f09d..0ade5e725d 100644
--- a/src/compiler/scheduler.h
+++ b/src/compiler/scheduler.h
@@ -6,11 +6,8 @@
 #define V8_COMPILER_SCHEDULER_H_
 
 #include "src/base/flags.h"
-#include "src/common/globals.h"
 #include "src/compiler/node.h"
-#include "src/compiler/opcodes.h"
 #include "src/compiler/schedule.h"
-#include "src/compiler/zone-stats.h"
 #include "src/zone/zone-containers.h"
 
 namespace v8 {
diff --git a/src/compiler/select-lowering.cc b/src/compiler/select-lowering.cc
index 6e96e4cef6..f398bf1101 100644
--- a/src/compiler/select-lowering.cc
+++ b/src/compiler/select-lowering.cc
@@ -5,10 +5,8 @@
 #include "src/compiler/select-lowering.h"
 
 #include "src/compiler/common-operator.h"
-#include "src/compiler/diamond.h"
 #include "src/compiler/graph-assembler.h"
 #include "src/compiler/graph.h"
-#include "src/compiler/node-properties.h"
 #include "src/compiler/node.h"
 
 namespace v8 {
diff --git a/src/compiler/simplified-lowering.cc b/src/compiler/simplified-lowering.cc
index 25b34c75c9..b13c4046a1 100644
--- a/src/compiler/simplified-lowering.cc
+++ b/src/compiler/simplified-lowering.cc
@@ -7,9 +7,8 @@
 #include <limits>
 
 #include "include/v8-fast-api-calls.h"
-#include "src/base/bits.h"
 #include "src/base/small-vector.h"
-#include "src/codegen/code-factory.h"
+#include "src/codegen/callable.h"
 #include "src/codegen/machine-type.h"
 #include "src/codegen/tick-counter.h"
 #include "src/compiler/access-builder.h"
@@ -30,7 +29,6 @@
 #include "src/compiler/type-cache.h"
 #include "src/numbers/conversions-inl.h"
 #include "src/objects/objects.h"
-#include "src/utils/address-map.h"
 
 #if V8_ENABLE_WEBASSEMBLY
 #include "src/wasm/value-type.h"
diff --git a/src/compiler/simplified-operator-reducer.cc b/src/compiler/simplified-operator-reducer.cc
index a6d8505f97..c0d5f0b4e4 100644
--- a/src/compiler/simplified-operator-reducer.cc
+++ b/src/compiler/simplified-operator-reducer.cc
@@ -12,7 +12,6 @@
 #include "src/compiler/opcodes.h"
 #include "src/compiler/operator-properties.h"
 #include "src/compiler/simplified-operator.h"
-#include "src/compiler/type-cache.h"
 #include "src/numbers/conversions-inl.h"
 
 namespace v8 {
diff --git a/src/compiler/simplified-operator-reducer.h b/src/compiler/simplified-operator-reducer.h
index 2f78d92016..97b3829afd 100644
--- a/src/compiler/simplified-operator-reducer.h
+++ b/src/compiler/simplified-operator-reducer.h
@@ -6,7 +6,6 @@
 #define V8_COMPILER_SIMPLIFIED_OPERATOR_REDUCER_H_
 
 #include "src/base/compiler-specific.h"
-#include "src/common/globals.h"
 #include "src/compiler/common-operator.h"
 #include "src/compiler/graph-reducer.h"
 
diff --git a/src/compiler/simplified-operator.cc b/src/compiler/simplified-operator.cc
index b0faaf9866..25b75f913e 100644
--- a/src/compiler/simplified-operator.cc
+++ b/src/compiler/simplified-operator.cc
@@ -10,11 +10,10 @@
 #include "src/compiler/opcodes.h"
 #include "src/compiler/operator.h"
 #include "src/compiler/types.h"
-#include "src/handles/handles-inl.h"
+#include "src/handles/handles-inl.h"  // for operator<<
 #include "src/objects/feedback-cell.h"
 #include "src/objects/map.h"
 #include "src/objects/name.h"
-#include "src/objects/objects-inl.h"
 
 #if V8_ENABLE_WEBASSEMBLY
 #include "src/compiler/wasm-compiler-definitions.h"
diff --git a/src/compiler/simplified-operator.h b/src/compiler/simplified-operator.h
index a5d9d771f6..f92a2f59b9 100644
--- a/src/compiler/simplified-operator.h
+++ b/src/compiler/simplified-operator.h
@@ -21,7 +21,6 @@
 #include "src/handles/handles.h"
 #include "src/handles/maybe-handles.h"
 #include "src/objects/objects.h"
-#include "src/objects/type-hints.h"
 #include "src/zone/zone-handle-set.h"
 
 namespace v8 {
diff --git a/src/compiler/state-values-utils.cc b/src/compiler/state-values-utils.cc
index 55073bc752..c5d75dc209 100644
--- a/src/compiler/state-values-utils.cc
+++ b/src/compiler/state-values-utils.cc
@@ -6,7 +6,6 @@
 
 #include "src/compiler/bytecode-liveness-map.h"
 #include "src/compiler/common-operator.h"
-#include "src/utils/bit-vector.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/state-values-utils.h b/src/compiler/state-values-utils.h
index 9c74ffdc46..db1dd81cf3 100644
--- a/src/compiler/state-values-utils.h
+++ b/src/compiler/state-values-utils.h
@@ -7,7 +7,6 @@
 
 #include <array>
 
-#include "src/common/globals.h"
 #include "src/compiler/common-operator.h"
 #include "src/compiler/js-graph.h"
 #include "src/zone/zone-hashmap.h"
diff --git a/src/compiler/store-store-elimination.cc b/src/compiler/store-store-elimination.cc
index c662103923..ad19f3f170 100644
--- a/src/compiler/store-store-elimination.cc
+++ b/src/compiler/store-store-elimination.cc
@@ -6,7 +6,6 @@
 
 #include "src/codegen/tick-counter.h"
 #include "src/compiler/all-nodes.h"
-#include "src/compiler/common-operator.h"
 #include "src/compiler/js-graph.h"
 #include "src/compiler/node-properties.h"
 #include "src/compiler/persistent-map.h"
diff --git a/src/compiler/type-cache.h b/src/compiler/type-cache.h
index 6442b6f6b0..00c3ba47fe 100644
--- a/src/compiler/type-cache.h
+++ b/src/compiler/type-cache.h
@@ -7,7 +7,6 @@
 
 #include "src/compiler/types.h"
 #include "src/date/date.h"
-#include "src/objects/code.h"
 #include "src/objects/js-array-buffer.h"
 #include "src/objects/string.h"
 
diff --git a/src/compiler/typed-optimization.h b/src/compiler/typed-optimization.h
index 8cfdb86d90..3ed9721958 100644
--- a/src/compiler/typed-optimization.h
+++ b/src/compiler/typed-optimization.h
@@ -6,7 +6,6 @@
 #define V8_COMPILER_TYPED_OPTIMIZATION_H_
 
 #include "src/base/compiler-specific.h"
-#include "src/common/globals.h"
 #include "src/compiler/graph-reducer.h"
 
 namespace v8 {
diff --git a/src/compiler/typer.cc b/src/compiler/typer.cc
index 611ffe7e19..3d6d83c314 100644
--- a/src/compiler/typer.cc
+++ b/src/compiler/typer.cc
@@ -19,7 +19,6 @@
 #include "src/compiler/operation-typer.h"
 #include "src/compiler/simplified-operator.h"
 #include "src/compiler/type-cache.h"
-#include "src/init/bootstrapper.h"
 #include "src/objects/objects-inl.h"
 
 namespace v8 {
diff --git a/src/compiler/typer.h b/src/compiler/typer.h
index d1b6afeaf9..710874de49 100644
--- a/src/compiler/typer.h
+++ b/src/compiler/typer.h
@@ -5,7 +5,6 @@
 #ifndef V8_COMPILER_TYPER_H_
 #define V8_COMPILER_TYPER_H_
 
-#include "src/common/globals.h"
 #include "src/compiler/graph.h"
 #include "src/compiler/operation-typer.h"
 
diff --git a/src/compiler/types.cc b/src/compiler/types.cc
index d333517a97..82fedec2cb 100644
--- a/src/compiler/types.cc
+++ b/src/compiler/types.cc
@@ -7,9 +7,8 @@
 #include <iomanip>
 
 #include "src/compiler/js-heap-broker.h"
-#include "src/handles/handles-inl.h"
+#include "src/numbers/conversions-inl.h"
 #include "src/objects/instance-type.h"
-#include "src/objects/objects-inl.h"
 #include "src/objects/turbofan-types.h"
 #include "src/utils/ostreams.h"
 
diff --git a/src/compiler/value-numbering-reducer.cc b/src/compiler/value-numbering-reducer.cc
index dfaa67b471..24ad97e6cc 100644
--- a/src/compiler/value-numbering-reducer.cc
+++ b/src/compiler/value-numbering-reducer.cc
@@ -6,7 +6,6 @@
 
 #include <cstring>
 
-#include "src/base/functional.h"
 #include "src/compiler/node-properties.h"
 #include "src/compiler/node.h"
 
diff --git a/src/compiler/verifier.cc b/src/compiler/verifier.cc
index 12598e8f8b..6086e1e068 100644
--- a/src/compiler/verifier.cc
+++ b/src/compiler/verifier.cc
@@ -20,11 +20,9 @@
 #include "src/compiler/operator-properties.h"
 #include "src/compiler/operator.h"
 #include "src/compiler/schedule.h"
-#include "src/compiler/simplified-operator.h"
 #include "src/compiler/state-values-utils.h"
 #include "src/compiler/type-cache.h"
 #include "src/utils/bit-vector.h"
-#include "src/utils/ostreams.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/verifier.h b/src/compiler/verifier.h
index 0be0102cc0..96c32a53c3 100644
--- a/src/compiler/verifier.h
+++ b/src/compiler/verifier.h
@@ -6,7 +6,6 @@
 #define V8_COMPILER_VERIFIER_H_
 
 #include "src/base/macros.h"
-#include "src/common/globals.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/wasm-compiler.cc b/src/compiler/wasm-compiler.cc
index 96182d9711..8a05c3c73a 100644
--- a/src/compiler/wasm-compiler.cc
+++ b/src/compiler/wasm-compiler.cc
@@ -6,17 +6,11 @@
 
 #include <memory>
 
-#include "src/api/api-inl.h"
 #include "src/base/optional.h"
-#include "src/base/platform/elapsed-timer.h"
-#include "src/base/platform/platform.h"
-#include "src/base/platform/wrappers.h"
 #include "src/base/small-vector.h"
 #include "src/base/v8-fallthrough.h"
 #include "src/base/vector.h"
-#include "src/codegen/assembler-inl.h"
 #include "src/codegen/assembler.h"
-#include "src/codegen/code-factory.h"
 #include "src/codegen/compiler.h"
 #include "src/codegen/interface-descriptors-inl.h"
 #include "src/codegen/machine-type.h"
@@ -39,12 +33,9 @@
 #include "src/compiler/pipeline.h"
 #include "src/compiler/wasm-compiler-definitions.h"
 #include "src/compiler/wasm-graph-assembler.h"
-#include "src/compiler/zone-stats.h"
-#include "src/execution/isolate-inl.h"
-#include "src/execution/simulator.h"
+#include "src/execution/simulator-base.h"
 #include "src/heap/factory.h"
 #include "src/logging/counters.h"
-#include "src/logging/log.h"
 #include "src/objects/heap-number.h"
 #include "src/objects/instance-type.h"
 #include "src/roots/roots.h"
diff --git a/src/compiler/wasm-compiler.h b/src/compiler/wasm-compiler.h
index b568124c3e..5a9b1f926b 100644
--- a/src/compiler/wasm-compiler.h
+++ b/src/compiler/wasm-compiler.h
@@ -15,7 +15,6 @@
 // Clients of this interface shouldn't depend on lots of compiler internals.
 // Do not include anything from src/compiler here!
 #include "src/base/small-vector.h"
-#include "src/objects/js-function.h"
 #include "src/runtime/runtime.h"
 #include "src/wasm/function-body-decoder.h"
 #include "src/wasm/function-compiler.h"
diff --git a/src/compiler/wasm-escape-analysis.cc b/src/compiler/wasm-escape-analysis.cc
index 450590d7d0..2a24bfb8d3 100644
--- a/src/compiler/wasm-escape-analysis.cc
+++ b/src/compiler/wasm-escape-analysis.cc
@@ -4,7 +4,7 @@
 
 #include "src/compiler/wasm-escape-analysis.h"
 
-#include "src/compiler/js-graph.h"
+#include "src/compiler/machine-graph.h"
 #include "src/compiler/node-properties.h"
 
 namespace v8 {
diff --git a/src/compiler/wasm-gc-lowering.cc b/src/compiler/wasm-gc-lowering.cc
index 03fa186b67..6c1f9e289f 100644
--- a/src/compiler/wasm-gc-lowering.cc
+++ b/src/compiler/wasm-gc-lowering.cc
@@ -7,11 +7,9 @@
 #include "src/base/logging.h"
 #include "src/common/globals.h"
 #include "src/compiler/common-operator.h"
-#include "src/compiler/node-matchers.h"
 #include "src/compiler/node-properties.h"
 #include "src/compiler/opcodes.h"
 #include "src/compiler/operator.h"
-#include "src/compiler/simplified-operator.h"
 #include "src/compiler/wasm-compiler-definitions.h"
 #include "src/compiler/wasm-graph-assembler.h"
 #include "src/wasm/object-access.h"
diff --git a/src/compiler/wasm-inlining.h b/src/compiler/wasm-inlining.h
index 8bdfd3dab8..01d7f2bca1 100644
--- a/src/compiler/wasm-inlining.h
+++ b/src/compiler/wasm-inlining.h
@@ -10,7 +10,7 @@
 #define V8_COMPILER_WASM_INLINING_H_
 
 #include "src/compiler/graph-reducer.h"
-#include "src/compiler/js-graph.h"
+#include "src/compiler/machine-graph.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/wasm-loop-peeling.cc b/src/compiler/wasm-loop-peeling.cc
index 65d87e3672..1c921ba751 100644
--- a/src/compiler/wasm-loop-peeling.cc
+++ b/src/compiler/wasm-loop-peeling.cc
@@ -4,11 +4,8 @@
 
 #include "src/compiler/wasm-loop-peeling.h"
 
-#include "src/base/small-vector.h"
-#include "src/codegen/tick-counter.h"
 #include "src/compiler/common-operator.h"
 #include "src/compiler/loop-analysis.h"
-#include "src/compiler/loop-peeling.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/wasm-loop-peeling.h b/src/compiler/wasm-loop-peeling.h
index 848dcceba0..2f6610f65a 100644
--- a/src/compiler/wasm-loop-peeling.h
+++ b/src/compiler/wasm-loop-peeling.h
@@ -10,7 +10,8 @@
 #define V8_COMPILER_WASM_LOOP_PEELING_H_
 
 #include "src/compiler/common-operator.h"
-#include "src/compiler/loop-analysis.h"
+#include "src/compiler/compiler-source-position-table.h"
+#include "src/compiler/node-origin-table.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/compiler/wasm-typer.cc b/src/compiler/wasm-typer.cc
index 5e99fd630b..187c0abc3e 100644
--- a/src/compiler/wasm-typer.cc
+++ b/src/compiler/wasm-typer.cc
@@ -9,7 +9,6 @@
 #include "src/compiler/node-matchers.h"
 #include "src/compiler/node-properties.h"
 #include "src/compiler/opcodes.h"
-#include "src/compiler/simplified-operator.h"
 #include "src/utils/utils.h"
 #include "src/wasm/object-access.h"
 #include "src/wasm/wasm-objects.h"
diff --git a/src/compiler/zone-stats.h b/src/compiler/zone-stats.h
index 0949c431b6..4b32f53ade 100644
--- a/src/compiler/zone-stats.h
+++ b/src/compiler/zone-stats.h
@@ -8,7 +8,6 @@
 #include <map>
 #include <vector>
 
-#include "src/common/globals.h"
 #include "src/zone/zone.h"
 
 namespace v8 {
diff --git a/src/init/setup-isolate-full.cc b/src/init/setup-isolate-full.cc
index 44b9f47efe..d927820ca6 100644
--- a/src/init/setup-isolate-full.cc
+++ b/src/init/setup-isolate-full.cc
@@ -2,13 +2,11 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "src/init/setup-isolate.h"
-
 #include "src/base/logging.h"
 #include "src/debug/debug-evaluate.h"
 #include "src/execution/isolate.h"
 #include "src/heap/heap-inl.h"
-#include "src/interpreter/interpreter.h"
+#include "src/init/setup-isolate.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/interpreter/bytecode-array-builder.h b/src/interpreter/bytecode-array-builder.h
index 13fc181c66..8cb099e51e 100644
--- a/src/interpreter/bytecode-array-builder.h
+++ b/src/interpreter/bytecode-array-builder.h
@@ -6,7 +6,6 @@
 #define V8_INTERPRETER_BYTECODE_ARRAY_BUILDER_H_
 
 #include "src/ast/ast.h"
-#include "src/base/compiler-specific.h"
 #include "src/base/export-template.h"
 #include "src/common/globals.h"
 #include "src/interpreter/bytecode-array-writer.h"
@@ -17,7 +16,6 @@
 #include "src/interpreter/bytecodes.h"
 #include "src/interpreter/constant-array-builder.h"
 #include "src/interpreter/handler-table-builder.h"
-#include "src/zone/zone-containers.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/interpreter/bytecode-array-iterator.h b/src/interpreter/bytecode-array-iterator.h
index af73a5e912..0d9e01583a 100644
--- a/src/interpreter/bytecode-array-iterator.h
+++ b/src/interpreter/bytecode-array-iterator.h
@@ -7,7 +7,6 @@
 
 #include <memory>
 
-#include "src/base/optional.h"
 #include "src/common/globals.h"
 #include "src/handles/handles.h"
 #include "src/interpreter/bytecode-register.h"
diff --git a/src/interpreter/bytecode-array-writer.cc b/src/interpreter/bytecode-array-writer.cc
index ab03873b4c..06c9e65372 100644
--- a/src/interpreter/bytecode-array-writer.cc
+++ b/src/interpreter/bytecode-array-writer.cc
@@ -9,7 +9,6 @@
 #include "src/interpreter/bytecode-jump-table.h"
 #include "src/interpreter/bytecode-label.h"
 #include "src/interpreter/bytecode-node.h"
-#include "src/interpreter/bytecode-register.h"
 #include "src/interpreter/bytecode-source-info.h"
 #include "src/interpreter/constant-array-builder.h"
 #include "src/interpreter/handler-table-builder.h"
diff --git a/src/interpreter/bytecode-array-writer.h b/src/interpreter/bytecode-array-writer.h
index 9976f59c23..672d544b39 100644
--- a/src/interpreter/bytecode-array-writer.h
+++ b/src/interpreter/bytecode-array-writer.h
@@ -5,7 +5,6 @@
 #ifndef V8_INTERPRETER_BYTECODE_ARRAY_WRITER_H_
 #define V8_INTERPRETER_BYTECODE_ARRAY_WRITER_H_
 
-#include "src/base/compiler-specific.h"
 #include "src/codegen/source-position-table.h"
 #include "src/common/globals.h"
 #include "src/interpreter/bytecodes.h"
diff --git a/src/interpreter/bytecode-decoder.h b/src/interpreter/bytecode-decoder.h
index 1bf93e092c..ea93dd3ded 100644
--- a/src/interpreter/bytecode-decoder.h
+++ b/src/interpreter/bytecode-decoder.h
@@ -7,9 +7,7 @@
 
 #include <iosfwd>
 
-#include "src/common/globals.h"
 #include "src/interpreter/bytecode-register.h"
-#include "src/interpreter/bytecodes.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/interpreter/bytecode-flags.cc b/src/interpreter/bytecode-flags.cc
index b2ed1dcecd..9a4a5cb1bd 100644
--- a/src/interpreter/bytecode-flags.cc
+++ b/src/interpreter/bytecode-flags.cc
@@ -6,7 +6,6 @@
 
 #include "src/ast/ast-value-factory.h"
 #include "src/ast/ast.h"
-#include "src/builtins/builtins-constructor.h"
 #include "src/objects/objects-inl.h"
 
 namespace v8 {
diff --git a/src/interpreter/bytecode-generator.cc b/src/interpreter/bytecode-generator.cc
index ae5f2a1ddb..e83e8aa811 100644
--- a/src/interpreter/bytecode-generator.cc
+++ b/src/interpreter/bytecode-generator.cc
@@ -29,10 +29,8 @@
 #include "src/logging/log.h"
 #include "src/numbers/conversions.h"
 #include "src/objects/debug-objects.h"
-#include "src/objects/literal-objects-inl.h"
-#include "src/objects/objects-inl.h"
 #include "src/objects/smi.h"
-#include "src/objects/template-objects-inl.h"
+#include "src/objects/template-objects.h"
 #include "src/parsing/parse-info.h"
 #include "src/parsing/token.h"
 #include "src/utils/ostreams.h"
diff --git a/src/interpreter/bytecode-generator.h b/src/interpreter/bytecode-generator.h
index 3e503a9a96..c3883de2a8 100644
--- a/src/interpreter/bytecode-generator.h
+++ b/src/interpreter/bytecode-generator.h
@@ -10,7 +10,6 @@
 #include "src/interpreter/bytecode-array-builder.h"
 #include "src/interpreter/bytecode-label.h"
 #include "src/interpreter/bytecode-register.h"
-#include "src/interpreter/bytecodes.h"
 #include "src/objects/feedback-vector.h"
 #include "src/objects/function-kind.h"
 
diff --git a/src/interpreter/bytecode-node.cc b/src/interpreter/bytecode-node.cc
index aa980ad713..dc8e4683d3 100644
--- a/src/interpreter/bytecode-node.cc
+++ b/src/interpreter/bytecode-node.cc
@@ -5,7 +5,6 @@
 #include "src/interpreter/bytecode-node.h"
 
 #include <iomanip>
-#include "src/codegen/source-position-table.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/interpreter/bytecode-node.h b/src/interpreter/bytecode-node.h
index ac6f5922f2..b73ddf0411 100644
--- a/src/interpreter/bytecode-node.h
+++ b/src/interpreter/bytecode-node.h
@@ -7,7 +7,6 @@
 
 #include <algorithm>
 
-#include "src/common/globals.h"
 #include "src/interpreter/bytecode-source-info.h"
 #include "src/interpreter/bytecodes.h"
 
diff --git a/src/interpreter/bytecode-register-allocator.h b/src/interpreter/bytecode-register-allocator.h
index 65dacbcbdf..953319da37 100644
--- a/src/interpreter/bytecode-register-allocator.h
+++ b/src/interpreter/bytecode-register-allocator.h
@@ -6,8 +6,6 @@
 #define V8_INTERPRETER_BYTECODE_REGISTER_ALLOCATOR_H_
 
 #include "src/interpreter/bytecode-register.h"
-#include "src/interpreter/bytecodes.h"
-#include "src/zone/zone-containers.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/interpreter/bytecode-register-optimizer.h b/src/interpreter/bytecode-register-optimizer.h
index 15aa17ca40..b3f910f08d 100644
--- a/src/interpreter/bytecode-register-optimizer.h
+++ b/src/interpreter/bytecode-register-optimizer.h
@@ -8,6 +8,8 @@
 #include "src/base/compiler-specific.h"
 #include "src/common/globals.h"
 #include "src/interpreter/bytecode-register-allocator.h"
+#include "src/zone/zone-containers.h"
+#include "src/zone/zone.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/interpreter/bytecode-source-info.cc b/src/interpreter/bytecode-source-info.cc
index bb95c1259a..1401e5d596 100644
--- a/src/interpreter/bytecode-source-info.cc
+++ b/src/interpreter/bytecode-source-info.cc
@@ -5,7 +5,6 @@
 #include "src/interpreter/bytecode-source-info.h"
 
 #include <iomanip>
-#include "src/codegen/source-position-table.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/interpreter/bytecodes.cc b/src/interpreter/bytecodes.cc
index e43c18c1a2..b0960ac050 100644
--- a/src/interpreter/bytecodes.cc
+++ b/src/interpreter/bytecodes.cc
@@ -6,7 +6,6 @@
 
 #include <iomanip>
 
-#include "src/base/bits.h"
 #include "src/interpreter/bytecode-traits.h"
 
 namespace v8 {
diff --git a/src/interpreter/constant-array-builder.cc b/src/interpreter/constant-array-builder.cc
index 260a75448d..8591a37183 100644
--- a/src/interpreter/constant-array-builder.cc
+++ b/src/interpreter/constant-array-builder.cc
@@ -9,10 +9,10 @@
 #include <set>
 
 #include "src/ast/ast-value-factory.h"
-#include "src/ast/ast.h"
 #include "src/ast/scopes.h"
 #include "src/base/functional.h"
 #include "src/execution/isolate.h"
+#include "src/handles/handles.h"
 #include "src/heap/local-factory-inl.h"
 #include "src/objects/objects-inl.h"
 
diff --git a/src/interpreter/constant-array-builder.h b/src/interpreter/constant-array-builder.h
index 2120142a12..14eec07edc 100644
--- a/src/interpreter/constant-array-builder.h
+++ b/src/interpreter/constant-array-builder.h
@@ -8,9 +8,8 @@
 #include "src/ast/ast-value-factory.h"
 #include "src/common/globals.h"
 #include "src/handles/handles.h"
-#include "src/interpreter/bytecodes.h"
+#include "src/interpreter/bytecode-operands.h"
 #include "src/objects/smi.h"
-#include "src/utils/identity-map.h"
 #include "src/zone/zone-containers.h"
 
 namespace v8 {
diff --git a/src/interpreter/handler-table-builder.h b/src/interpreter/handler-table-builder.h
index 8670fc0492..d74bb65616 100644
--- a/src/interpreter/handler-table-builder.h
+++ b/src/interpreter/handler-table-builder.h
@@ -7,7 +7,6 @@
 
 #include "src/codegen/handler-table.h"
 #include "src/interpreter/bytecode-register.h"
-#include "src/interpreter/bytecodes.h"
 #include "src/objects/fixed-array.h"
 #include "src/zone/zone-containers.h"
 
diff --git a/src/interpreter/interpreter-assembler.cc b/src/interpreter/interpreter-assembler.cc
index 4212882875..0ab530934a 100644
--- a/src/interpreter/interpreter-assembler.cc
+++ b/src/interpreter/interpreter-assembler.cc
@@ -10,11 +10,9 @@
 #include "src/codegen/code-factory.h"
 #include "src/codegen/interface-descriptors-inl.h"
 #include "src/codegen/machine-type.h"
-#include "src/execution/frames.h"
 #include "src/interpreter/bytecodes.h"
 #include "src/interpreter/interpreter.h"
 #include "src/objects/objects-inl.h"
-#include "src/zone/zone.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/interpreter/interpreter-assembler.h b/src/interpreter/interpreter-assembler.h
index 32239012ef..bce2fa5674 100644
--- a/src/interpreter/interpreter-assembler.h
+++ b/src/interpreter/interpreter-assembler.h
@@ -5,13 +5,11 @@
 #ifndef V8_INTERPRETER_INTERPRETER_ASSEMBLER_H_
 #define V8_INTERPRETER_INTERPRETER_ASSEMBLER_H_
 
-#include "src/builtins/builtins.h"
 #include "src/codegen/code-stub-assembler.h"
 #include "src/common/globals.h"
 #include "src/interpreter/bytecode-register.h"
 #include "src/interpreter/bytecodes.h"
 #include "src/runtime/runtime.h"
-#include "src/utils/allocation.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/interpreter/interpreter-generator.cc b/src/interpreter/interpreter-generator.cc
index 033fadf618..717b4c4672 100644
--- a/src/interpreter/interpreter-generator.cc
+++ b/src/interpreter/interpreter-generator.cc
@@ -8,10 +8,7 @@
 #include <tuple>
 
 #include "src/builtins/builtins-constructor-gen.h"
-#include "src/builtins/builtins-iterator-gen.h"
 #include "src/builtins/profile-data-reader.h"
-#include "src/codegen/code-factory.h"
-#include "src/debug/debug.h"
 #include "src/ic/accessor-assembler.h"
 #include "src/ic/binary-op-assembler.h"
 #include "src/ic/ic.h"
diff --git a/src/interpreter/interpreter-intrinsics-generator.cc b/src/interpreter/interpreter-intrinsics-generator.cc
index abb574f9e1..c211631e18 100644
--- a/src/interpreter/interpreter-intrinsics-generator.cc
+++ b/src/interpreter/interpreter-intrinsics-generator.cc
@@ -5,16 +5,11 @@
 #include "src/interpreter/interpreter-intrinsics-generator.h"
 
 #include "src/builtins/builtins.h"
-#include "src/codegen/code-factory.h"
-#include "src/execution/frames.h"
 #include "src/heap/factory-inl.h"
-#include "src/interpreter/bytecodes.h"
 #include "src/interpreter/interpreter-assembler.h"
 #include "src/interpreter/interpreter-intrinsics.h"
 #include "src/objects/js-generator.h"
 #include "src/objects/objects-inl.h"
-#include "src/objects/source-text-module.h"
-#include "src/utils/allocation.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/interpreter/interpreter.cc b/src/interpreter/interpreter.cc
index b9ccae9a26..cad488ab32 100644
--- a/src/interpreter/interpreter.cc
+++ b/src/interpreter/interpreter.cc
@@ -15,17 +15,13 @@
 #include "src/common/globals.h"
 #include "src/execution/local-isolate.h"
 #include "src/heap/parked-scope.h"
-#include "src/init/bootstrapper.h"
 #include "src/init/setup-isolate.h"
 #include "src/interpreter/bytecode-generator.h"
 #include "src/interpreter/bytecodes.h"
 #include "src/logging/runtime-call-stats-scope.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/shared-function-info.h"
-#include "src/objects/slots.h"
-#include "src/objects/visitors.h"
 #include "src/parsing/parse-info.h"
-#include "src/snapshot/snapshot.h"
 #include "src/utils/ostreams.h"
 
 namespace v8 {
diff --git a/src/interpreter/interpreter.h b/src/interpreter/interpreter.h
index 82fc8a9dea..1efe5ee74a 100644
--- a/src/interpreter/interpreter.h
+++ b/src/interpreter/interpreter.h
@@ -13,11 +13,11 @@
 #include "src/base/macros.h"
 #include "src/builtins/builtins.h"
 #include "src/interpreter/bytecodes.h"
-#include "src/runtime/runtime.h"
 
 namespace v8 {
 namespace internal {
 
+class AccountingAllocator;
 class BytecodeArray;
 class Callable;
 class UnoptimizedCompilationJob;
diff --git a/src/json/json-parser.h b/src/json/json-parser.h
index a4fa8fd552..7a0711623c 100644
--- a/src/json/json-parser.h
+++ b/src/json/json-parser.h
@@ -12,7 +12,6 @@
 #include "src/execution/isolate.h"
 #include "src/heap/factory.h"
 #include "src/objects/objects.h"
-#include "src/zone/zone-containers.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/json/json-stringifier.cc b/src/json/json-stringifier.cc
index ef56a645c1..cbc951d88a 100644
--- a/src/json/json-stringifier.cc
+++ b/src/json/json-stringifier.cc
@@ -15,7 +15,6 @@
 #include "src/objects/ordered-hash-table.h"
 #include "src/objects/smi.h"
 #include "src/strings/string-builder-inl.h"
-#include "src/utils/utils.h"
 
 namespace v8 {
 namespace internal {
diff --git a/src/libplatform/default-foreground-task-runner.cc b/src/libplatform/default-foreground-task-runner.cc
index 85487c16d6..5aaf4d1713 100644
--- a/src/libplatform/default-foreground-task-runner.cc
+++ b/src/libplatform/default-foreground-task-runner.cc
@@ -5,7 +5,7 @@
 #include "src/libplatform/default-foreground-task-runner.h"
 
 #include "src/base/platform/mutex.h"
-#include "src/libplatform/default-platform.h"
+#include "src/base/platform/time.h"
 
 namespace v8 {
 namespace platform {
diff --git a/src/libplatform/default-platform.cc b/src/libplatform/default-platform.cc
index 748a2cf9e7..ecac542b82 100644
--- a/src/libplatform/default-platform.cc
+++ b/src/libplatform/default-platform.cc
@@ -8,7 +8,6 @@
 #include <queue>
 
 #include "include/libplatform/libplatform.h"
-#include "src/base/bounded-page-allocator.h"
 #include "src/base/debug/stack_trace.h"
 #include "src/base/logging.h"
 #include "src/base/page-allocator.h"
diff --git a/src/libplatform/default-platform.h b/src/libplatform/default-platform.h
index 0d49ffd0ed..4e746fbabb 100644
--- a/src/libplatform/default-platform.h
+++ b/src/libplatform/default-platform.h
@@ -13,9 +13,7 @@
 #include "include/libplatform/v8-tracing.h"
 #include "include/v8-platform.h"
 #include "src/base/compiler-specific.h"
-#include "src/base/macros.h"
 #include "src/base/platform/mutex.h"
-#include "src/base/platform/time.h"
 
 namespace v8 {
 namespace platform {
diff --git a/src/libplatform/delayed-task-queue.h b/src/libplatform/delayed-task-queue.h
index 135ef53efe..8420bbe8e5 100644
--- a/src/libplatform/delayed-task-queue.h
+++ b/src/libplatform/delayed-task-queue.h
@@ -10,7 +10,6 @@
 #include <queue>
 
 #include "include/libplatform/libplatform-export.h"
-#include "src/base/macros.h"
 #include "src/base/platform/condition-variable.h"
 #include "src/base/platform/mutex.h"
 
diff --git a/src/libplatform/task-queue.h b/src/libplatform/task-queue.h
index 55748a382e..57defb10f9 100644
--- a/src/libplatform/task-queue.h
+++ b/src/libplatform/task-queue.h
@@ -9,7 +9,6 @@
 #include <queue>
 
 #include "include/libplatform/libplatform-export.h"
-#include "src/base/macros.h"
 #include "src/base/platform/mutex.h"
 #include "src/base/platform/semaphore.h"
 #include "testing/gtest/include/gtest/gtest_prod.h"  // nogncheck
diff --git a/src/libplatform/worker-thread.h b/src/libplatform/worker-thread.h
index 8dae4777b7..1607c99d3f 100644
--- a/src/libplatform/worker-thread.h
+++ b/src/libplatform/worker-thread.h
@@ -9,7 +9,6 @@
 
 #include "include/libplatform/libplatform-export.h"
 #include "src/base/compiler-specific.h"
-#include "src/base/macros.h"
 #include "src/base/platform/platform.h"
 
 namespace v8 {
diff --git a/src/wasm/jump-table-assembler.h b/src/wasm/jump-table-assembler.h
index 9d8ff58803..f5aeabee53 100644
--- a/src/wasm/jump-table-assembler.h
+++ b/src/wasm/jump-table-assembler.h
@@ -9,6 +9,7 @@
 #ifndef V8_WASM_JUMP_TABLE_ASSEMBLER_H_
 #define V8_WASM_JUMP_TABLE_ASSEMBLER_H_
 
+#include "src/codegen/flush-instruction-cache.h"
 #include "src/codegen/macro-assembler.h"
 
 namespace v8 {
diff --git a/test/cctest/compiler/code-assembler-tester.h b/test/cctest/compiler/code-assembler-tester.h
index d1e904a6a8..c2a5a6db6e 100644
--- a/test/cctest/compiler/code-assembler-tester.h
+++ b/test/cctest/compiler/code-assembler-tester.h
@@ -5,12 +5,13 @@
 #ifndef V8_TEST_CCTEST_COMPILER_CODE_ASSEMBLER_TESTER_H_
 #define V8_TEST_CCTEST_COMPILER_CODE_ASSEMBLER_TESTER_H_
 
+#include "src/codegen/assembler.h"
 #include "src/codegen/interface-descriptors.h"
 #include "src/compiler/code-assembler.h"
 #include "src/compiler/raw-machine-assembler.h"
 #include "src/execution/isolate.h"
 #include "src/handles/handles.h"
-#include "test/cctest/compiler/function-tester.h"
+#include "test/cctest/cctest.h"
 
 namespace v8 {
 namespace internal {
diff --git a/test/cctest/compiler/test-codegen.h b/test/cctest/compiler/test-codegen.h
index cf9c0a2c57..108c67a2fb 100644
--- a/test/cctest/compiler/test-codegen.h
+++ b/test/cctest/compiler/test-codegen.h
@@ -5,11 +5,12 @@
 #ifndef V8_CCTEST_COMPILER_CODEGEN_TESTER_H_
 #define V8_CCTEST_COMPILER_CODEGEN_TESTER_H_
 
+#include "src/codegen/assembler.h"
 #include "src/codegen/optimized-compilation-info.h"
 #include "src/compiler/backend/instruction-selector.h"
 #include "src/compiler/pipeline.h"
 #include "src/compiler/raw-machine-assembler.h"
-#include "src/execution/simulator.h"
+#include "src/objects/code-inl.h"
 #include "test/cctest/cctest.h"
 #include "test/cctest/compiler/call-tester.h"
 
@@ -372,7 +373,6 @@ class CompareWrapper {
       default:
         UNREACHABLE();
     }
-    return nullptr;
   }
 
   bool Int32Compare(int32_t a, int32_t b) {
@@ -390,7 +390,6 @@ class CompareWrapper {
       default:
         UNREACHABLE();
     }
-    return false;
   }
 
   bool Float64Compare(double a, double b) {
@@ -404,7 +403,6 @@ class CompareWrapper {
       default:
         UNREACHABLE();
     }
-    return false;
   }
 
   IrOpcode::Value opcode;
diff --git a/test/cctest/compiler/test-run-calls-to-external-references.cc b/test/cctest/compiler/test-run-calls-to-external-references.cc
index 3d31d8e59e..5b7d7d077f 100644
--- a/test/cctest/compiler/test-run-calls-to-external-references.cc
+++ b/test/cctest/compiler/test-run-calls-to-external-references.cc
@@ -2,8 +2,8 @@
 // source code is governed by a BSD-style license that can be found in the
 // LICENSE file.
 
+#include "src/base/memory.h"
 #include "src/codegen/external-reference.h"
-#include "src/objects/objects-inl.h"
 #include "test/cctest/cctest.h"
 #include "test/cctest/compiler/test-codegen.h"
 #include "test/cctest/compiler/value-helper.h"
diff --git a/test/common/assembler-tester.h b/test/common/assembler-tester.h
index 7a9e360925..880d983967 100644
--- a/test/common/assembler-tester.h
+++ b/test/common/assembler-tester.h
@@ -8,8 +8,8 @@
 #include <memory>
 
 #include "src/codegen/assembler.h"
-#include "src/codegen/code-desc.h"
-#include "src/common/code-memory-access-inl.h"
+#include "src/codegen/flush-instruction-cache.h"
+#include "src/common/code-memory-access.h"
 
 namespace v8 {
 namespace internal {
-- 
2.35.1

